modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewDXPKey/index.js

```diff
@@ -0,0 +1,318 @@
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+/* eslint-disable no-unused-vars */
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+import ClayAlert from '@clayui/alert';
+import {ClaySelect} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import React, {useEffect, useState} from 'react';
+import {Link} from 'react-router-dom';
+import {Button} from '../../../../common/components';
+import {Radio} from '../../../../common/components/Radio';
+import Layout from '../../../../common/containers/setup-forms/Layout';
+import {useApplicationProvider} from '../../../../common/context/AppPropertiesProvider';
+import {getNewGenerateKeyFormValues} from '../../../../common/services/liferay/rest/raysource/LicenseKeys';
+import getCurrentEndDate from '../../../../common/utils/getCurrentEndDate';
+import {useCustomerPortal} from '../../context';
+import {PAGE_TYPES, PRODUCT_TYPES} from '../../utils/constants';
+import GenerateNewKeySkeleton from './Skeleton';
+
+const ACTIVATION_ROOT_ROUTER = 'activation';
+const INICIAL_SELECTED_VERSION = '7.0';
+const INICIAL_SELECTED_TYPE = 'Backup';
+
+const GenerateNewDXPKey = ({accountKey, productGroupName, sessionId}) => {
+	const {licenseKeyDownloadURL} = useApplicationProvider();
+	const [{subscriptionGroups}] = useCustomerPortal();
+
+	const [selected, setSelected] = useState();
+	const [generateFormValues, setGenerateFormValues] = useState();
+	const [selectedVersion, setSelectedVersion] = useState(
+		INICIAL_SELECTED_VERSION
+	);
+	const [selectedKeyType, setSelectedKeyType] = useState(
+		INICIAL_SELECTED_TYPE
+	);
+
+	const handleComeBackPage = `/${accountKey}/${ACTIVATION_ROOT_ROUTER}/${PAGE_TYPES.dxp}`;
+
+	useEffect(() => {
+		const fetchGenerateFormData = async () => {
+			const data = await getNewGenerateKeyFormValues(
+				accountKey,
+				licenseKeyDownloadURL,
+				PRODUCT_TYPES.dxp,
+				sessionId
+			);
+
+			if (data) {
+				setGenerateFormValues(data);
+			}
+		};
+		fetchGenerateFormData();
+	}, [accountKey, licenseKeyDownloadURL, productGroupName, sessionId]);
+
+	const productVersions = generateFormValues?.versions?.sort((a, b) =>
+		a.label - b.label > 0 ? 1 : -1
+	);
+
+	const productTypes = generateFormValues?.versions?.map(
+		(version) => version.types
+	);
+
+	const productKeyTypes = productTypes?.map((productType) =>
+		productType
+			.map((licenseKey) =>
+				licenseKey.licenseEntryName.replace('DXP ', '')
+			)
+			.sort()
+	);
+
```

Comment:
I think you can do this directly:
```suggestion
	const productKeyTypes = generateFormValues?.versions?.map(({types}) =>
		types
			.map((licenseKey) =>
				licenseKey.licenseEntryName.replace(DXP_VARIABLE_HERE, '')
			)
			.sort()
	);

```
Also, declare a variable for simple DXP text